### PR TITLE
feat: add arbitrum goerli

### PR DESCRIPTION
**Pull request by gzeoneth**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

This PR add support for Arbitrum Goerli Testnet (chainId: 421613)

- **Other information**:

The Arbitrum Team have deployed an instance of Uniswap V3 on Arbitrum Goerli as follow
```
const ARBITRUM_GOERLI_V3_CORE_FACTORY_ADDRESSES = '0x4893376342d5D7b3e31d4184c08b265e5aB2A3f6'
const ARBITRUM_GOERLI_ROUTER_ADDRESS = '0xab7664500b19a7a2362Ab26081e6DfB971B6F1B0'
const ARBITRUM_GOERLI_V3_MIGRATOR_ADDRESSES = '0xA815919D2584Ac3F76ea9CB62E6Fd40a43BCe0C3'
const ARBITRUM_GOERLI_MULTICALL_ADDRESS = '0x8260CB40247290317a4c062F3542622367F206Ee'
const ARBITRUM_GOERLI_QUOTER_ADDRESSES = '0x1dd92b83591781D0C6d98d07391eea4b9a6008FA'
const ARBITRUM_GOERLI_NONFUNGIBLE_POSITION_MANAGER_ADDRESSES = '0x622e4726a167799826d1E1D150b076A7725f5D81'
const ARBITRUM_GOERLI_TICK_LENS_ADDRESSES = '0xb52429333da969a0C79a60930a4Bf0020E5D1DE8'
```

Dependent on https://github.com/Uniswap/smart-order-router/pull/188

### CODE REVIEW

File: src/components/DowntimeWarning/index.tsx [hunk 1]
```diff
@@ -64,6 +64,7 @@ export default function DowntimeWarning() {
       )
     case SupportedChainId.ARBITRUM_ONE:
     case SupportedChainId.ARBITRUM_RINKEBY:
+    case SupportedChainId.ARBITRUM_GOERLI:
```
Comment (vm) [inline src/components/DowntimeWarning/index.tsx:67]: this was deleted, so you can rebase against main to update

File: src/constants/tokens.ts [hunk 2]
```diff
@@ -74,6 +74,13 @@ export const USDC_ARBITRUM_RINKEBY = new Token(
   'USDC',
   'USD//C'
 )
+export const USDC_ARBITRUM_GOERLI = new Token(
```
Comment (vm) [inline src/constants/tokens.ts:77]: after you rebase you may need to remove the export on this const

**Review by vm — APPROVED** [review]
after conflicts are fixed this LGTM

**Comment by vm** [comment]
will merge after the smart order router change is merged

**Comment by gzeoneth** [comment]
> will merge after the smart order router change is merged

ready to merge

**Comment by gzeoneth** [comment]
Need https://github.com/Uniswap/sdk-core/pull/36 to fix the optional cypress test

File: yarn.lock [hunk 3]
```diff
@@ -5103,6 +5143,15 @@
     "@uniswap/v2-core" "1.0.1"
     "@uniswap/v3-core" "1.0.0"
 
+"@uniswap/universal-router@^1.0.1":
```
Comment (zzmp) [inline yarn.lock:5146]: This must coalesce with 1.2.1.

File: yarn.lock [hunk 4]
```diff
@@ -5028,10 +5029,22 @@
     tiny-invariant "^1.1.0"
     toformat "^2.0.0"
 
+"@uniswap/sdk-core@^3.1.1":
```
Comment (zzmp) [inline yarn.lock:5032]: This must coalesce with the entry above it.

File: yarn.lock [hunk 5]
```diff
@@ -5028,10 +5029,22 @@
...
+  integrity sha512-afXn3R2ICf1mAZU+Ug8MmPbuXG7PYjoUYr0+sKLApyOPuhlxpwV4tR4ba8w4jSXoFhqOWxZ9qh44isWoScK96w==
+  dependencies:
+    "@ethersproject/address" "^5.0.2"
+    big.js "^5.2.2"
+    decimal.js-light "^2.5.0"
+    jsbi "^3.1.4"
+    tiny-invariant "^1.1.0"
+    toformat "^2.0.0"
+
 "@uniswap/smart-order-router@^2.10.0":
```
Comment (zzmp) [inline yarn.lock:5044]: This should go away.

**Review by zzmp — CHANGES_REQUESTED** [review]

**Review by zzmp — CHANGES_REQUESTED** [review]
manifest/lockfile changes should be made in a separate PR to make them easier to review and reduce the chance of dependency version skew (eg this introduces a version mismatch with sdk-core).

This PR should wait until https://github.com/Uniswap/interface/pull/5806 is merged, and then merge that in to take care of the package/lockfile updates.

File: patches/@uniswap+smart-order-router+2.10.0.patch [hunk 6]
```diff
@@ -0,0 +1,738 @@
...
+ export declare const USDC_POLYGON: Token;
+diff --git a/node_modules/@uniswap/smart-order-router/build/main/providers/token-provider.js b/node_modules/@uniswap/smart-order-router/build/main/providers/token-provider.js
+index a164918..6ca1834 100644
+--- a/node_modules/@uniswap/smart-order-router/build/main/providers/token-provider.js
++++ b/node_modules/@uniswap/smart-order-router/build/main/providers/token-provider.js
+@@ -56,6 +56,7 @@ exports.DAI_ARBITRUM_RINKEBY = new sdk_core_1.Token(util_1.ChainId.ARBITRUM_RINK
+ exports.USDT_ARBITRUM_RINKEBY = new sdk_core_1.Token(util_1.ChainId.ARBITRUM_RINKEBY, '0x920b9301c2de92186299cd2abc7199e25b9728b3', 6, 'UDST', 'Tether USD');
+ exports.USDC_ARBITRUM_RINKEBY = new sdk_core_1.Token(util_1.ChainId.ARBITRUM_RINKEBY, '0x09b98f8b2395d076514037ff7d39a091a536206c', 6, 'USDC', 'USD//C');
+ exports.UNI_ARBITRUM_RINKEBY = new sdk_core_1.Token(util_1.ChainId.ARBITRUM_RINKEBY, '0x049251a7175071316e089d0616d8b6aacd2c93b8', 18, 'UNI', 'Uni token');
++exports.USDC_ARBITRUM_GOERLI = new sdk_core_1.Token(util_1.ChainId.ARBITRUM_GOERLI, '0x8FB1E3fC51F3b789dED7557E680551d93Ea9d892', 6, 'USDC', 'USD//C'); // This is the AAVE Test USDC
```
Comment (vm) [inline patches/@uniswap+smart-order-router+2.10.0.patch:35]: what is this line doing?

Comment (gzeoneth) [reply patches/@uniswap+smart-order-router+2.10.0.patch]: This add the ArbGoerli version of USDC, the comment is incorrect and removed

**Review by vm — APPROVED** [review]

**Pull request merged**